shopfloor/actions/stock.py

```diff
@@ -35,6 +35,22 @@ def mark_move_line_as_picked(
         )
         move_lines.picking_id.filtered(lambda p: p.user_id != user).user_id = user.id
 
+    def unmark_move_line_as_picked(self, move_lines):
+        """Reverse the change from `mark_move_line_as_picked`."""
+        move_lines.write(
+            {
+                "shopfloor_user_id": False,
+                "qty_done": 0,
+                "result_package_id": False,
+            }
+        )
+        move_lines.move_id.picking_id.write(
```

Comment:
1/ move lines could be not linked to moves, you should take the picking_id from the move_lines directly
2/ you should check that there are no other move lines in the picking that still have a shopfloor_user_id, otherwise you should extract current move lines in a split order with `_extract_in_split_order`. This will ensure that we don't end-up with a picking without user_id still having move lines with a shopfloor_user_id